src/components/ModalWindow/AddModal/AddModal.tsx

```diff
@@ -0,0 +1,175 @@
+  Form,
+  TitleModal,
+  InputDate,
+  InputLeft,
+  InputRight,
+  Label,
+  StyledFlex,
+  StyledFlexForButtons,
+  TextArea
+} from "./AddModal.styled";
+import {PrimaryButton, SecondaryButton} from "../../Button/Button.styled";
+import {genreOptions} from "../../../assets/data/constData";
+import {backgroundGrey, backgroundMain, baseWeight, mainFontColor, xlarge, red} from "../../../styles/global_varables";
+import {Multiselect} from "multiselect-react-dropdown";
+
+interface ModalWindowProps {
+  form: {
+    title: string,
+    releaseDate: string,
+    url: string,
+    rating: string | number,
+    genre: { value: string, label: string }[],
+    runtime: string | number,
+    overview: string
+  };
+  handleOnChange: (e: SyntheticEvent) => void;
+  handleSubmit: (e: SyntheticEvent) => void;
+  handleFormReset: (e: SyntheticEvent) => void;
+  handleGenreChange: (selectedList: [], selectedItem: {}) => void;
+  mode: string;
+}
+
+const AddModalWindow: React.FC<ModalWindowProps> = (props) => {
+  const {form, handleOnChange, handleSubmit, handleFormReset, handleGenreChange, mode} = props;
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <TitleModal>{mode === 'add' ? 'Add movie' : 'Edit movie'}</TitleModal>
+      <StyledFlex>
+        <div>
+          <Label htmlFor="title">Title</Label>
+          <InputLeft
+            type="text"
+            name="title"
+            id="title"
+            placeholder="Title"
+            value={form.title}
+            onChange={handleOnChange}
+          />
+        </div>
+        <div>
+          <Label htmlFor="releaseDate">Release Date</Label>
+          <InputDate
+            type="date"
+            name="releaseDate"
+            id="releaseDate"
+            value={form.releaseDate}
+            onChange={handleOnChange}
+          />
+        </div>
+      </StyledFlex>
+      <StyledFlex>
+        <div>
+          <Label htmlFor="url">Models URL</Label>
+          <InputLeft
+            type="text"
+            name="url"
+            id="url"
+            placeholder="https://"
+            value={form.url}
+            onChange={handleOnChange}
+          />
+        </div>
+        <div>
+          <Label htmlFor="rating">Rating</Label>
+          <InputRight
+            type="number"
+            name="rating"
+            id="rating"
+            min="0"
+            max="10"
+            step="0.1"
+            value={form.rating}
+            onChange={handleOnChange}
+          />
+        </div>
+      </StyledFlex>
+      <StyledFlex>
+        <div>
+          <Label htmlFor="genre">Genre</Label>
+          <Multiselect
+            displayValue="label"
+            selectedValues={form.genre}
+            hidePlaceholder={true}
+            placeholder='Select Genre'
+            onRemove={handleGenreChange}
+            onSelect={handleGenreChange}
+            options={genreOptions}
+            showCheckbox
+            showArrow={false}
+            style={{
```

Comment:
Impressive styling. You can extract this `Multiselect` into a separate component to isolate this crazy css.